src/models/StyleSheet.js

```diff
@@ -117,18 +117,32 @@ export default class StyleSheet {
       return existingTag
     }
 
-    const lastTag = this.tags[this.tags.length - 1]
-    const componentTag = (!lastTag || lastTag.isFull() || lastTag.isLocal !== isLocal)
-      ? this.createNewTag(isLocal)
-      : lastTag
-    this.componentTags[componentId] = componentTag
-    componentTag.addComponent(componentId)
-    return componentTag
+    let tag = this.tags.find(t => t.isLocal === isLocal && !t.isFull())
```

Comment:
Won't this line find previous local tags that aren't full and append them there? Effectively maintaining two "last tags", one for local, and one for global styles? Then they both get appended until they're full and more get created?